src/main.cpp

```diff
@@ -3220,6 +3240,7 @@ bool CBlock::ConnectBlock(CTxDB& txdb, CBlockIndex* pindex, bool fJustCheck, boo
 
 
             //If we are out of sync, and research age is enabled, and the superblock is valid, load it now, so we can continue checking blocks accurately
+            // I would suggest to NOT bother with superblock at all here. It will be loaded in tally.
```

Comment:
What about this? I do not like loading superblock at two places (determinism). Either load it in tally only, which I feel would be better, or only on block and startup.

Reply:
I think we should load it on accept and when reorganizing. If we only load it when tallying we won't see the new SB until the tally window has been moved past or onto it, so there will still be multiple superblocks in succession. Unless I've missed something.

Reply:
Tally runs on reorg, right, so it should pick up sb in window.
For double superblocks, I previously added scan of last few blocks to Need superblock function, which affects miner. In this PR i made it to reject sb if not needed.